openehr-terminology/src/main/java/com/nedap/archie/terminology/OpenEHRTerminologyAccess.java

```diff
@@ -19,8 +19,10 @@
 
 public class OpenEHRTerminologyAccess implements TerminologyAccess {
 
-    static volatile OpenEHRTerminologyAccess instance;
+    private static final Pattern openEHRTermIdPattern = Pattern.compile("http://openehr.org/id/(?<id>[0-9]+)");
+    private static final Pattern IANATermIdPattern = Pattern.compile("https://www.w3.org/ns/iana/media-types/(?<type>[A-Za-z]+)/(?<subtype>[A-Za-z]+)#Resource");
```

Comment:
This regex is a bit too restrictive. Media types can also contain numbers and other special characters, see https://github.com/openEHR/archie/blob/master/openehr-terminology/src/main/resources/openEHR_RM/openehr_external_terminologies.xml#L525